packages/noya-renderer/src/components/SketchFileRenderer.tsx

```diff
@@ -169,6 +170,72 @@ export default memo(function SketchFileRenderer() {
     [page, state.selectedObjects],
   );
 
+  const getSelectedAndHighlightedLayerBoundingRec = useCallback(() => {
+    if (
+      state.selectedObjects.length === 0 ||
+      !highlightedLayer ||
+      !boundingRect
+    ) {
+      return;
+    }
+
+    const highlightedBoundingRec = getBoundingRect(
+      page,
+      AffineTransform.identity,
+      [highlightedLayer.id],
+      {
+        clickThroughGroups: true,
+        includeHiddenLayers: true,
+      },
+    );
+    return {
+      selectedBoundingRec: boundingRect,
+      highlightedBoundingRec: highlightedBoundingRec,
+    };
+  }, [boundingRect, highlightedLayer, page, state.selectedObjects.length]);
+
+  const distanceLabelAndPathBetweenSketchLayers = useMemo(() => {
+    if (
+      !highlightedLayer ||
+      state.selectedObjects.includes(highlightedLayer.id)
+    ) {
+      return;
+    }
+
+    const indexPath = findIndexPath(
+      page,
+      (layer) => layer.do_objectID === highlightedLayer.id,
+    );
+
+    if (!indexPath) return;
+
+    let selectedAndHighlightedLayers;
+    if (highlightedLayer.isMeasured) {
+      selectedAndHighlightedLayers = getSelectedAndHighlightedLayerBoundingRec();
+    }
+
+    return (
+      highlightedLayer && (
```

Comment:
We already check for `!highlightedLayer` above, so we can remove this check. We also shouldn't need a `<>` here. I think if we combine some of the checks into the condition above, we shouldn't need any `&&` in this return

Reply:
Because `getBoundingRect` returns possibly an undefined value I can't find a way around adding one check in the return for `selectedAndHighlightedLayers.highlightedBoundingRec`. The other check have been removed.